pyrogram/client/ext/utils.py

```diff
@@ -711,6 +711,22 @@ def parse_messages(
     return parsed_messages if is_list else parsed_messages[0]
 
 
+def parse_deleted_messages(
+        messages: list,
+        channel_id: int
+) -> pyrogram_types.Messages:
+    parsed_messages = []
+
+    for message in messages:
+        m = pyrogram_types.Message(
+            message_id=message,
+            chat=(pyrogram_types.Chat(id=channel_id, type="channel") if channel_id is not None else None)
```

Comment:
Here we must use `id=int("-100" + str(channel_id))` to convert the channel_id into its Bot API id representation.